nucleus/payara-modules/asadmin-audit/src/main/java/fish/payara/audit/AdminAuditService.java

```diff
@@ -0,0 +1,203 @@
+        return auditLevel;
+    }
+    
+    public void setAuditLevel(AuditLevel level){
+        auditLevel = level;
+    }
+    
+    /**
+     * Starts all notifiers that have been enable with the admin audit service.
+     */
+    public synchronized void bootstrapNotifierList() {
+        notifierExecutionOptionsList = new ArrayList<>();
+        if (configuration.getNotifierList() != null) {
+            for (Notifier notifier : configuration.getNotifierList()) {
+                ConfigView view = ConfigSupport.getImpl(notifier);
+                NotifierConfigurationType annotation = view.getProxyType().getAnnotation(NotifierConfigurationType.class);
+                NotifierExecutionOptionsFactory<Notifier> factory = executionOptionsFactoryStore.get(annotation.type());
+                if (factory != null) {
+                    notifierExecutionOptionsList.add(factory.build(notifier));
+                }
+            }
+        }
+        if (notifierExecutionOptionsList.isEmpty()) {
+            // Add logging execution options by default
+            LogNotifierExecutionOptions logNotifierExecutionOptions = new LogNotifierExecutionOptions();
+            logNotifierExecutionOptions.setEnabled(true);
+            notifierExecutionOptionsList.add(logNotifierExecutionOptions);
+        }
+    }
+    
+    /**
+     * Gets a list of all the options of all notifiers configured with the asadmin audit service.
+     * @return 
+     */
+    public List<NotifierExecutionOptions> getNotifierExecutionOptionsList() {
+        return notifierExecutionOptionsList;
+    }
+    
+    
+    public void recordAsadminCommand(String command, ParameterMap parameters, Subject subject) {
+        if (enabled && notifierExecutionOptionsList != null && checkAuditLevel(command)) {
+            
+            Set<Principal> principals = subject.getPrincipals();
+            String name = principals.iterator().next().getName();
+            for (NotifierExecutionOptions notifierExecutionOptions : notifierExecutionOptionsList) {
+                
+                if (notifierExecutionOptions.isEnabled()) {
+                    NotificationEventFactory notificationEventFactory = eventFactoryStore.get(notifierExecutionOptions.getNotifierType());
+                    NotificationEvent notificationEvent = notificationEventFactory
+                            .buildNotificationEvent(Level.WARNING, AUDIT_MESSAGE, name + " issued command " + command + " with parameters " + parameters.toString(), null);
+                    notificationSevice.notify(EventSource.AUDIT, notificationEvent);
+                }
+            }
+        }
+    }
+    
+    private boolean checkAuditLevel(String command) {        
+        switch (auditLevel) {
+            case INTERNAL:
+                return true;
+            case ACCESSORS:
+                if (!command.startsWith("_")) {
+                    return true;
+                }
+                break;
```

Comment:
rather than the break here, you could just return false. Easier to follow the code then